Add Vulkan backend to llama demo binary

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #2608

## Context

Cleans up Vulkan backend CMake build so that `vulkan_backend` target is a single static library will all required symbols. Then add vulkan backend to llama runner binary.

Differential Revision: [D55272794](https://our.internmc.facebook.com/intern/diff/D55272794)

SS-JIA committed:
Add Vulkan backend to llama demo binary

[ghstack-poisoned]

SS-JIA commented:
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55272794).

mcr229 reviewed (approved):
looks good but I think you'll need to rebase for merge conflicts on my changes

SS-JIA committed:
Update on "Add Vulkan backend to llama demo binary"


## Context

Cleans up Vulkan backend CMake build so that `vulkan_backend` target is a single static library will all required symbols. Then add vulkan backend to llama runner binary.

Differential Revision: [D55272794](https://our.internmc.facebook.com/intern/diff/D55272794)

[ghstack-poisoned]

SS-JIA commented:
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55272794).
